### PR TITLE
More preparations for TOPPView refactor

### PR DESCRIPTION
**Pull request by cbielow**
## Description

this PR pulls in all the non-GUI changes to OpenMS.
E.g. AreaIterator now also considers IonMobility when iterating over spectra. It also used named arguments for easier configuration.

Along the way, small fixes, e.g. to trivially copy Peak2D (was not possible before...)

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/KERNEL/MSExperiment.cpp [hunk 1]
```diff
@@ -777,6 +818,19 @@ namespace OpenMS
...
   }
 
+  bool MSExperiment::isIMFrame() const
+  {
+    if (spectra_.empty()) return false;
+    auto rt_start = spectra_[0].getRT();
+    auto last_drift = std::numeric_limits<double>::lowest();
+    for (const auto& s : spectra_) {
+      if (s.getRT() != rt_start) return false; // RT changes...
+      if (s.getDriftTime() == last_drift) return false; // IM did not change...
```
Comment (enetz) [inline src/openms/source/KERNEL/MSExperiment.cpp:828]: this disallows equal IM values in IMFrames. Is it not possible to have two peaks with the same IM by collapsing a range of m/z values? or is it assumed here, that some kind of filtering for equal values has been performed for those cases?

**Review by enetz — APPROVED** [review]
looks good to merge, after tests pass
(abibuilder and jenkins tests seem to be down for now)

**Comment by enetz** [comment]
rebuild jenkins

**Comment by cbielow** [comment]
test pass, except Windows due to Sirius (`Error: Unexpected internal error (Sirius was executed, but an empty output was generated)` -- I guess unrelated?

**Comment by timosachsenberg** [comment]
Yes unrelated

**Pull request merged**